rs/nervous_system/canisters/src/registry.rs

```diff
@@ -91,144 +88,231 @@ impl Registry for RegistryCanister {
     }
 }
 
-type FakeGetChangesSince = BTreeMap<u64, Result<Vec<RegistryDelta>, Error>>;
-type FakeGetLatestVersion = Vec<Result<u64, Error>>;
-
-/// This uses a Vec instead of VecDeque since inserting test values won't necessarily happen
-/// in the same order as it would in the real registry.
-type FakeRegistryMap = BTreeMap<Vec<u8>, Vec<RegistryValue>>;
-
-/// A fake registry client for testing.  This re-implements some internals from the Registry
-/// as it is easier to interact with this store as though you were making Registry updates
-/// than trying to mock out responses.  Additionally, it is possible to override the responses
-/// so that you can test errors.  Responses that are explicitly set are consumed.
-#[derive(Default)]
-pub struct FakeRegistry {
-    version: AtomicU64,
-    store: Arc<Mutex<FakeRegistryMap>>,
-    override_get_changes_since: Arc<Mutex<FakeGetChangesSince>>,
-    override_get_latest_version: Arc<Mutex<FakeGetLatestVersion>>,
-}
+/// Test version of the Registry trait.  This is used to mock out the registry for testing
+/// in a non-wasm environment.  This is not used in production and is only used for testing.
+#[cfg(not(target_arch = "wasm32"))]
```

Comment:
Thanks for marking this module as non-canister code. 

I recall that at some point (maybe still) we used some registry client in the production replica tooling, and that was quite unfortunate. I wonder if we should mark this module as test-only explicitly to avoid something like that happening in the future. I leave it up to you to decide, just wanted to point out that not-wasm != test-only.